File: Data Structures & Algorithms/combination-target-sum/submission-2.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        ans = []
        def dfs(arr,val,curr):
            if val == target:
                ans.append(arr.copy())
                return
            if val > target:
                return 
            for i in range(curr,len(nums)):
                arr.append(nums[i])
                dfs(arr,val+ nums[i], i)
                arr.pop()
        dfs([],0,0)
        return ans
        

File: Data Structures & Algorithms/remove-element/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} val
     * @return {number}
     */
    removeElement(nums, val) {
        let i = 0, j = nums.length-1;
        while(i<=j){
           if(nums[i] === val){
                nums[i] = nums[j--];
           }else i++;
        }
        return j+1;
    }
}
